### PR TITLE
#306 - Added support for minimum ruby version

### PR DESCRIPTION
**Pull request by emilianodellacasa**
For issue #306 I added support to a minimum ruby version 2.3.3 ad discussed in the same issue.

I have also modified the configuration for rubocop, travis and appveyor

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#336](https://github.com/zold-io/zold/issues/336) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#336](https://github.com/zold-io/zold/issues/336) is assigned to @oltdaniel[/z](https://www.0crat.com/u/oltdaniel), here is [why](https://www.0crat.com/footprint/CAZPZR9FS/1841539767); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by emilianodellacasa** [comment]
@oltdaniel ruby 2.6 apparently is not supported by appveyor

https://www.appveyor.com/docs/build-environment/#ruby

**Comment by emilianodellacasa** [comment]
@yegor256 Apparently there is a conflict in a file, may you solve the conflict?

**Comment by yegor256** [comment]
@emilianodellacasa only you can solve the conflict :) It's your branch. Let me know when it's ready and I will try to merge.

**Comment by emilianodellacasa** [comment]
@yegor256 Solved, sorry for asking but I thought I was missing some privilege or something like that! :)

File: .rubocop.yml [hunk 1]
```diff
@@ -33,3 +33,5 @@ Layout/MultilineOperationIndentation:
   Enabled: false
 Naming/UncommunicativeMethodParamName:
   MinNameLength: 1
+Style/FrozenStringLiteralComment:
```
Comment (yegor256) [inline .rubocop.yml]: @emilianodellacasa what this is for? Can't you just do what Rubocop is suggesting?

File: zold.gemspec [hunk 2]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |s|
   s.specification_version = 2 if s.respond_to? :specification_version=
   s.required_rubygems_version = Gem::Requirement.new('>= 0') if s.respond_to? :required_rubygems_version=
   s.rubygems_version = '2.2'
-  s.required_ruby_version = '~>2.2'
+  s.required_ruby_version = '>=   2.3.3'
```
Comment (yegor256) [inline zold.gemspec]: @emilianodellacasa I think we have too many spaces here :)

**Comment by yegor256** [comment]
@emilianodellacasa see a few comments above

**Comment by emilianodellacasa** [comment]
@yegor256 ping

File: Gemfile [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (yegor256) [inline Gemfile]: @emilianodellacasa what is this? You should do `ruby `>= 2.3.3'.freeze`

**Comment by yegor256** [comment]
@emilianodellacasa there is some mess with these frozen string literals. What's going on, can you explain? Why can't you use `.freeze` where necessary?

**Comment by emilianodellacasa** [comment]
Starting from Ruby 3.0 all constant freeze will be frozen, so since Ruby 2.3 the magic comment frozen_string_literal has been introduced to help the transition. Without that comment rubocop will complain.

References:
https://stackoverflow.com/questions/37799296/what-does-the-comment-frozen-string-literal-true-do
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/FrozenStringLiteralComment

**Comment by 0crat** [comment]
@oltdaniel[/z](https://www.0crat.com/u/oltdaniel) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by emilianodellacasa** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor mertge

**Comment by emilianodellacasa** [comment]
@yegor256 Please merge again

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@emilianodellacasa seems to be a problem.. did you test it locally?

**Comment by emilianodellacasa** [comment]
@yegor256 Locally it is working, that is what it's strange to me. Moreover, codeclimate and travis are giving no problem at all, this is the reason why I opened an issue on rultor.

Now I am trying a different way to specify the ruby version on Gemfile using the pessimistic operator, could you please try merging again?

**Comment by oltdaniel** [comment]
@emilianodellacasa @yegor256 `Gemfile` cannot understand the `>=` operator. Had the same issue, but haven't requested the change before my vacation. Sorry.

**Comment by emilianodellacasa** [comment]
@yegor256 Solved conflict

**Comment by emilianodellacasa** [comment]
@oltdaniel That is crazy, thanks for confirming

**Comment by yegor256** [comment]
@emilianodellacasa again merge conflicts :(

**Comment by emilianodellacasa** [comment]
@yegor256 Conflicts solved - which next? :)

**Comment by emilianodellacasa** [comment]
@yegor256 please try merging again

**Comment by emilianodellacasa** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@yegor256 To avoid being stuck on this PR too long (see https://github.com/yegor256/rultor/issues/1267) I would suggest to keep the Gemfile version where Rultor doesn't complain, and change Travis configuration to only check against Ruby version 2.3.3.

Then I will open a new issue with the necessary modifications in order to support ruby version >= 2.3.3 when the rultor issue will be solved.

**Comment by yegor256** [comment]
@emilianodellacasa seems fine for me. Again, keep in mind, it's not "Rultor issue". It may be Ubuntu issue, but has nothing to do with Rultor. Rultor just runs everything in a Docker container. Run it locally in the same container and you will see the same problem.

**Comment by emilianodellacasa** [comment]
@yegor256 I modified travis configuration, now both Travis and codeclimate are failing because of other commits that have been merged. 

Could you please merge?

**Comment by yegor256** [comment]
@rultor merge

**Comment by ypshenychka** [comment]
@oltdaniel According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your [message](https://github.com/zold-io/zold/pull/336#pullrequestreview-134598713) by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@emilianodellacasa According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your [message](https://github.com/zold-io/zold/pull/336#issuecomment-402963940) by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@oltdaniel According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by emilianodellacasa** [comment]
@ypshenychka Fixed, thanks a lot

**Comment by ypshenychka** [comment]
@emilianodellacasa thank you!

**Comment by ypshenychka** [comment]
@oltdaniel ping

**Comment by emilianodellacasa** [comment]
@ypshenychka @oltdaniel is on vacation now https://github.com/zold-io/zold/pull/347#discussion_r202057676

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @oltdaniel[/z](https://www.0crat.com/u/oltdaniel)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**